table_cache_test.go

```diff
@@ -323,8 +323,13 @@ func TestVirtualReadsWiring(t *testing.T) {
 	v2.LargestPointKey = v2.Largest
 	v2.SmallestPointKey = v2.Smallest
 
-	v1.ValidateVirtual(parentFile)
-	v2.ValidateVirtual(parentFile)
+	internalIterator, _, err := d.newIters(context.TODO(), v1, nil, internalIterOpts{})
+	require.NoError(t, err)
+	v1.ValidateVirtual(parentFile, internalIterator, d.cmp)
```

Comment:
It'd be best to pass `d.newIters` into `ValidateVirtual`, so we only incur the cost of creating the iterators if invariants are enabled.

Reply:
yeah I was trying that as well but there is some issues because some of the arguments such as `internalIterOpts` is not exported (and used in a lot of places), also packages such as `keyspan` and `pebble` cause import issues if used in the `manifest` package. Also, I think some places such as `sstable/reader_test.go:317:26` do not have a `db` object, not sure if I should try creating one or is there an alternative.

Reply:
I would solve that by passing in an adapter version of `newIters` to `ValidateVirtual` that doesn't take the `internalIterOpts` field, instead passing in the zero value for it. And that test can probably just pass a `nil` value for `newIters` and `ValidateVirtual` can make a check for whether that function is non-nil before calling it.

Reply:
yeah I tried doing that but when typing the new function inside of `ValidateVirtual` I had issues since I can't use the `keyspan` module (which is needed to type `keyspan.FragmentIterator`.

Reply:
yeah I tried doing that, but the issue I was facing was how to type the return value of the new function, specifically `keyspan.FragmentIterator` since I don't think I can import the `keyspan` module.